tapir/coop/models.py

```diff
@@ -719,3 +718,63 @@ def populate(
             old_frozen=old_frozen,
             new_frozen=new_frozen,
         )
+
+class ResignedMembership(models.Model):
```

Comment:
Maybe `MembershipResignation` is a bit better?

Reply:
Your IDE may be able to do the renaming for you. In PyCharm under the right-click menu I have Refactor/Rename, yours probably has something similar?

You'll have to re-do the migrations yes, you can delete the files that have been generated before and run `makemigrations` again. The content of your DB will then be invalid, but you can reset it (delete the Docker container and create it again) to start fresh.

Reply:
@Theophile-Madet What do you mean "the files that have been generated before"? These migrations-files, in which "ResignedMembership is in"? Only delete them or all migrations-files in the coop app? If you could answer me this question, then I'll do a new commit after I did the makemigrations here on my local. Thanks so much for your patience with me ;)

Reply:
We need to keep the migration files that existed before your changes and delete the ones that come from your changes.
This way, when you run `makemigrations` again, Django will generate new files with the new names.

Reply:
I'll try it

Reply:
makemiragtions and migrate worked, but resetting the test-data gives this traceback:
`django.db.utils.IntegrityError: update or delete on table "log_logentry" violates foreign key constraint "coop_resignmembershi_logentry_ptr_id_37732a13_fk_log_logen" on table "coop_resignmembershipcreatelogentry"
DETAIL:  Key (id)=(2600) is still referenced from table "coop_resignmembershipcreatelogentry".`

Any ideas?

Reply:
It tries to delete a `ResignMembership`, but that object is still referenced in a `resignmembershipcreatelogentry`.
I'm guessing that in `tapir.utils.management.commands.generate_test_data_functions.clear_data`, when it deletes the members, the corresponding `ResignMembership` objects should get deleted too but that doesn't work because of the reference in the log entry.

Maybe try adding your log entry class in `clear_data`, somewhere above `ShareOwner`?

Reply:
I deleted the hole docker container. Now it's working.